src/bindings/lua/jansson-lua.c

```diff
@@ -136,32 +134,32 @@ static int lua_is_integer (lua_State *L, int index)
     return (0);
 }
 
-int lua_value_to_json (lua_State *L, int i, json_object **valp)
+int lua_value_to_json (lua_State *L, int i, json_t **valp)
 {
     int index = (i < 0) ? (lua_gettop (L) + 1) + i : i;
-    json_object *o = NULL;
+    json_t *o = NULL;
 
     if (lua_isnoneornil (L, i))
         return (-1);
 
     switch (lua_type (L, index)) {
         case LUA_TNUMBER:
             if (lua_is_integer (L, index))
-                o = json_object_new_int64 (lua_tointeger (L, index));
+                o = json_integer (lua_tointeger (L, index));
```

Comment:
Each of these json_object functions can return NULL.  Should that be checked for and luaL_error() be used to set an error?  (Not sure - since result is  NULL, maybe that is sufficient?)

Reply:
The NULL should be sufficient, but the function would return success (`0`) even if `o` was NULL. Will fix.

Reply:
BTW, meant to mention that `luaL_error()` could also work here, but that throws an error in the lua interpreter and immediately stops the script. The current method is to return `nil, error_message` to the caller, which would potentially allow the script to recover or issue a different error message, etc. I'm actually kind of ambivalent which way to go, but since the existing behavior was to return error to the caller, we can keep it that way for now and rethink the approach when the bindings are updated.